state/protocol/protocol_state/kvstore_storage.go

```diff
@@ -36,7 +34,7 @@ type ProtocolKVStore interface {
 	//
 	// Expected errors of the returned anonymous function:
 	//   - storage.ErrAlreadyExists if a KV store for the given blockID has already been indexed.
-	IndexTx(blockID flow.Identifier, stateID flow.Identifier) func(*transaction.Tx) error
+	BatchIndex(rw storage.ReaderBatchWriter, blockID flow.Identifier, stateID flow.Identifier) error
```

Comment:
same here: the intention was to precompute a bunch of stuff and then commit the changes in the end.